modules/runtime/nuxeo-runtime/src/main/java/org/nuxeo/runtime/RuntimeMessageHandlerImpl.java

```diff
@@ -86,56 +89,43 @@ public void beforeDeactivation(ComponentManager mgr) {
     }
 
     protected void changeStep(ComponentManagerStep step) {
-        messages.removeIf(msg -> step.equals(msg.getStep()));
-        this.step = step;
+        this.currentStep = step;
     }
 
-    /**
-     * @since 9.10
-     */
-    protected static class Message {
-
-        protected final ComponentManagerStep step;
-
-        protected final Level level;
-
-        protected final String message;
-
-        public Message(ComponentManagerStep step, Level level, String message) {
-            this.step = step;
-            this.level = level;
-            this.message = message;
-        }
-
-        public ComponentManagerStep getStep() {
-            return step;
-        }
-
-        public Level getLevel() {
-            return level;
-        }
-
-        public String getMessage() {
-            return message;
-        }
-
+    @Override
+    public void addMessage(RuntimeMessage message) {
+        messages.add(message);
     }
 
-    /**
-     * @since 9.10
-     */
-    protected enum ComponentManagerStep {
-
-        ACTIVATING,
-
-        STARTING,
+    @Override
+    public void addMessage(Level level, String message, String source) {
```

Comment:
Does it make sense to have instead/in addition an api with Class<? > as source type?

Reply:
Instead makes sens given the usage :+1:

Reply:
Will check the DeploymentPreprocessor use case (only usage based on a String for now)

Reply:
Made the changed and used the addMessage(RuntimeMessage) api for the DeploymentPreprocessor use case (as nuxeo-runtime cannot depend on nuxeo-runtime-deploy module without producing a dependency cycle)